css-text-4/Overview.bs

```diff
@@ -31,8 +35,519 @@ Introduction</h2>
+		and risk causing overflow.
+		<pre><code class=lang-css>
+		@supports (word-boundary-detection: auto(ja)) {
+			h1:lang(ja) {
+				word-boundary-detection: auto(ja);
+				word-break: keep-all;
+			}
+		}
+		</code></pre>
+	</div>
+
+	[=Virtual word boundary=] insertion happens before [[CSS-TEXT-3#white-space-phase-1]]
+	and before [[#word-boundary-expansion]].
+	Later operations
+	(including [[CSS-TEXT-3#white-space-rules]], [=line breaking=], and [=intrinsic sizing=])
+	must take the presence of the [=virtual word boundary=] into account.
+	[=Selectors=] are not affected.
+
+	Inline box boundaries
+	and out-of-flow elements must be ignored
+	when determining word boundaries.
+
+	If a word boundary is found at the same position as
+	one or more inline box boundaries,
+	the [=virtual word boundary=] must be inserted
+	in the outermost element that participates in this inline box boundary.
+
+	<div class=example>
+		In the following example,
+		the red “<code><span style="color:red">|</span></code>” indicates
+		reasonable positions for a User Agent to insert virtual word boundaries:
+		<pre><code highlight=html>กรุงเทพ<span style="color:red">|</span>คือ<span style="color:red">|</span>สวยงาม</code></pre>
+		If that sentence had contained some inline markup,
+		the following example shows the correct position to insert the virtual word boundaries:
+		<pre><code highlight=html>กรุงเทพ<span style="color:red">|</span>คือ<span style="color:red">|</span>&lt;em>สวยงาม&lt;/em></code></pre>
+		The following example shows <em>incorrect</em> positions:
+		<pre><code highlight=html>กรุงเทพ<span style="color:red">|</span>คือ&lt;em><span style="color:red">|</span>สวยงาม&lt;/em></code></pre>
+		The following shows the correct positions in a more contrieved situation:
+		<pre><code highlight=html>กรุงเทพ<span style="color:red">|</span>&lt;b>&lt;u>คือ&lt;/u><span style="color:red">|</span>&lt;em>สวยงาม&lt;/em>&lt;/b></code></pre>
+	</div>
+
+	The User Agent may tailor its word boundary detection algorithm
+	depending on whether 'line-break' is
+	''loose''/''line-break/normal''/''line-break/strict''.
+
+	The User Agent must not insert a [=virtual word boundary=]:
+	<ul>
+		<li>
+			at the beginning or end of any box
+			(including [=inline boxes=])
+			whose parent box has a [=used value=]
+			of ''word-boundary-detection/manual''.
+
+		<li>
+			immediately adjacent to a [=word-separator character=],
+			or an [=other space separator=],
+			or a ZERO WIDTH SPACE (U+200B) character.
+
+			Note: This implies that for languages such as English
+			where words are separated by spaces or other separating characters,
+			''word-boundary-detection/auto(<lang>)'' has no effect.
+
+		<li>
+			between a [=typographic letter unit=]
+			and a subsequent [=typographic character unit=] from the [[!UNICODE]] Pe or Pf classes,
+			or between a [=typographic letter unit=]
+			and a preceeding [=typographic character unit=] from the [[!UNICODE]] Ps or Pi classes,
```

Comment:
IIRC Pf/Pi classes are characters that are sometimes used as opening and sometimes used as closing, so this might not be quite correct.

Reply:
Good point. I've though some more about it, and derived updated rules from UAX14 line breaking classes, rather than directly from unicode general categories.